### PR TITLE
feat: add Support for High-FPS Readings in Flashlight for Devices Over 60 FPS

### PR DESCRIPTION
**Pull request by MalcolmTomisin**
## Description
This PR introduces a new feature to Flashlight that extends support for devices capable of displaying frame rates (FPS) higher than 60Hz. It addresses this [issue](https://github.com/bamlab/flashlight/issues/135) 

## Key Changes:
High-FPS Support: Adjustments have been made to ensure that FPS readings above 60Hz are correctly captured and displayed by the flashlight tool.
Data-model: A `DeviceSpecs` type has been created, including the refresh rate into the `Results` type didn't seem like an optimal way to model the results of profiling.
Testing: Added unit tests to verify the functionality of the `adb` command for fps information.
Minor: Extracting socket events into enums

## Challenges
There's a lot of prop drilling going on, perhaps, including the refresh rate into the results type could have avoided all the prop drilling, looking forward to comments on that.

#### Additional Notes
- This update is backward-compatible with devices that have a refresh rate of 60Hz or below.
- No significant impact on existing functionality or performance.
- Further improvements could come through a `DeviceSpecs` type. Adding device specs of the device being profiled could be helpful metadata to add to reports.

#### Screenshots
image
image

### CODE REVIEW

File: packages/commands/measure/src/__tests__/__snapshots__/measure.test.tsx.snap [hunk 1]
```diff
@@ -12,7 +12,11 @@ Time taken to run the test.
 Can be helpful to measure Time To Interactive of your app, if the test is checking app start for instance.
 Average FPS
 60 FPS
-Frame Per Second. Your app should display 60 Frames Per Second to give an impression of fluidity. This number should be close to 60, otherwise it will seem laggy. 
+Frame Per Second. Your app should display 
+60
```
Comment (MalcolmTomisin) [inline packages/commands/measure/src/__tests__/__snapshots__/measure.test.tsx.snap:16]: Head scratching moment why this change happened

**Review by Almouro — COMMENTED** [review]
Hi @MalcolmTomisin !

First of all, apologies for the delay answering and thanks so much for submitting this great PR!! ❤️
This has been an open issue for some time now, and it's so cool to see someone spending a lot of time to fix it! 🙏

I've added some comments here and there, but I have to ask: what is the reasoning behind not adding the refresh rate in the results object? 
It seems to me that adding the device specs directly there would simplify the code. 

Also `flashlight` relies in a lot of place on a simple `json` file being the source of truth for a report. For instance, `flashlight test` outputs a json that can be used by `flashlight report`, and `flashlight measure` can also output one. `flashlight cloud` also has a similar mechanism.
So putting the device specs out of it might make it tricky to have for instance:
- flashlight test recognizing 120FPS and outputting results in json file
- flashlight report displaying the report for the json file, taking into account that the json file was a result for a 120FPS phone

I'll be away again for 10 days without access to computer, but I'm very happy to discuss this by message over Discord/Twitter/... if you want to get this in as soon as possible 🤞

**Comment by MalcolmTomisin** [comment]
> Hi @MalcolmTomisin !
> 
> First of all, apologies for the delay answering and thanks so much for submitting this great PR!! ❤️ This has been an open issue for some time now, and it's so cool to see someone spending a lot of time to fix it! 🙏
> 
> I've added some comments here and there, but I have to ask: what is the reasoning behind not adding the refresh rate in the results object? It seems to me that adding the device specs directly there would simplify the code.
> 
> Also `flashlight` relies in a lot of place on a simple `json` file being the source of truth for a report. For instance, `flashlight test` outputs a json that can be used by `flashlight report`, and `flashlight measure` can also output one. `flashlight cloud` also has a similar mechanism. So putting the device specs out of it might make it tricky to have for instance:
> 
> * flashlight test recognizing 120FPS and outputting results in json file
> * flashlight report displaying the report for the json file, taking into account that the json file was a result for a 120FPS phone
> 
> I'll be away again for 10 days without access to computer, but I'm very happy to discuss this by message over Discord/Twitter/... if you want to get this in as soon as possible 🤞

I’ve considered adding the `refreshRate` to the results JSON, but I wasn’t sure how to structure it in a way that maintains backward compatibility, I also want to avoid duplicating the refreshRate across multiple report arrays. You had suggested adding it to both TestCaseResult and AveragedTestCaseResult, but since the refreshRate wouldn’t change during a profiling session, I felt it might be redundant.

Additionally, I’m thinking about including device specifications in the reports, which I believe would be a significant benefit for transparency when reporting performance data.

File: packages/platforms/android/src/commands/__tests__/detectCurrentDeviceRefreshRate.test.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  it("retrieves correctly device refresh rate", () => {
+    executeCommandSpy.mockImplementation((command) => {
+      expect(command).toEqual(
+        'adb shell dumpsys display | grep -E "mRefreshRate|DisplayDeviceInfo"'
+      );
+
+      return sampleOutput;
+    });
+
+    expect(detectCurrentDeviceRefreshRate()).toEqual(60);
```
[the remark this answers is not in the document]
Comment (MalcolmTomisin) [reply packages/platforms/android/src/commands/__tests__/detectCurrentDeviceRefreshRate.test.ts:18]: I have added a fix to accommodate the configurations on Pixel

File: packages/platforms/android/src/commands/atrace/pollFpsUsage.ts [hunk 3]
```diff
@@ -27,9 +28,9 @@ export const parseLine = (
   };
 };
 
-// At some point we might want to change this to adapt to 90fps or 120fps devices
-const TARGET_FRAME_RATE = 60;
+const TARGET_FRAME_RATE = refreshRateManager.getRefreshRate();
```
[the remark this answers is not in the document]
Comment (MalcolmTomisin) [reply packages/platforms/android/src/commands/atrace/pollFpsUsage.ts:31]: Did I address your questions entirely?

[the remark this answers is not in the document]
Comment (MalcolmTomisin) [reply packages/platforms/android/src/commands/atrace/pollFpsUsage.ts:31]: This has been my understanding image
The refresh rate is retrieved 
- to set a target frame rate when FPS usage is polled
- passed to the web reporter UI over a socket for display and calculation purposes

**Comment by MalcolmTomisin** [comment]
> We could add a `deviceSpecs` field to both `TestCaseResult` and `AveragedTestCaseResult` Yeah I agree it's a bit weird to have it in both, `AveragedTestCaseResult` is a bit of a weird object at the moment to be fair and already duplicates the `name` of the report for instance. It would just get copied over right here https://github.com/bamlab/flashlight/blob/main/packages/core/reporter/src/reporting/averageIterations.ts#L76
> 
> We could make the field accept an undefined value for backwards compatibility.
> 
> What do you think?

Sounds fine to me.

File: packages/platforms/android/src/commands/__tests__/detectCurrentDeviceRefreshRate.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Almouro) [reply packages/platforms/android/src/commands/__tests__/detectCurrentDeviceRefreshRate.test.ts:19]: Awesome, thanks!

File: packages/platforms/android/src/commands/atrace/pollFpsUsage.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Almouro) [reply packages/platforms/android/src/commands/atrace/pollFpsUsage.ts:31]: Right, I see, makes sense

**Comment by Almouro** [comment]
Thanks @MalcolmTomisin for the updates!
Looks really good to me and seems to be working very well 👌

I'll probably get a second look and we should release this early next week! 🥳
